### ISSUE TITLE
onnx export wrong output dim

### ISSUE THREAD

**Issue by Slyne** [pytorch/pytorch#71408]
### 🐛 Describe the bug

I have a model and one of its output dimensions is different in pytorch 1.9 vs  > pytorch==1.10.0a0+0aef44c (this version is from nvidia's pytorch 21.10 docker).  Pytorch version greater than the NVIDIA's docker all produce wrong output. Haven't tried 1.11 latest build yet.

![image](https://user-images.githubusercontent.com/6286804/149970606-e0dcb9b2-8c9a-47f4-a86b-d973adf1438e.png)
The red circle should be 256.  This encoder_out tensor is after a layer norm operation.

This is the right (pytorch 1.9) model output:
![image](https://user-images.githubusercontent.com/6286804/149972916-3a8b65ad-0ddd-4036-9b00-e737fce952e6.png)

Dont know why after 1.10

## To reproduce
1. Follow here to install environment: https://github.com/wenet-e2e/wenet#installation
2. Then go to ```cd examples/aishell2/s0```
3. Download a model & uncompress it: 
```wget http://mobvoi-speech-public.ufile.ucloud.cn/public/wenet/aishell2/20210618_u2pp_conformer_exp.tar.gz
tar zxvf 20210618_u2pp_conformer_exp.tar.gz
```
4.  Export to onnx:
```
. ./path.sh

exp_dir=20210618_u2pp_conformer_exp
onnx_dir=u2pp_conformer_onnx

python3 wenet/bin/export_onnx.py --config=$exp_dir/train.yaml --checkpoint=$exp_dir/final.pt --cmvn_file=$exp_dir/global_cmvn --ctc_weight=0.1 --reverse_weight=0.4 --output_onnx_dir=$onnx_dir 
```
5. Check the model in ```u2pp_conformer_onnx```

### Versions

PyTorch version: 1.10.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.21.3
Libc version: glibc-2.31

Python version: 3.8.12 | packaged by conda-forge | (default, Sep 29 2021, 19:52:28)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-142-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.4.120
GPU models and configuration:
GPU 0: NVIDIA Quadro P6000
GPU 1: NVIDIA Quadro P6000

Nvidia driver version: 465.19.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.2
[pip3] nvidia-dlprof-pytorch-nvtx==1.6.0
[pip3] pytorch-quantization==2.1.0
[pip3] torch==1.10.1
[pip3] torchaudio==0.10.1
[pip3] torchtext==0.11.0a0
[pip3] torchvision==0.11.2
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.5                      281    conda-forge
[conda] mkl-include               2019.5                      281    conda-forge
[conda] numpy                     1.21.2           py38he2449b9_0    conda-forge
[conda] nvidia-dlprof-pytorch-nvtx 1.6.0                    pypi_0    pypi
[conda] pytorch-quantization      2.1.0                    pypi_0    pypi
[conda] torch                     1.10.1                   pypi_0    pypi
[conda] torchaudio                0.10.1                   pypi_0    pypi
[conda] torchtext                 0.11.0a0                 pypi_0    pypi
[conda] torchvision               0.11.2                   pypi_0    pypi

**Comment by rosrad** [comment]
I have met this similar issue as well. The onnx export output wrong dynamic axes which should be fixed.

**Comment by garymm** [comment]
I hope this was fixed by https://github.com/pytorch/pytorch/commit/3f02ad09ecca4576cfc00d1b4b42a2a928770132 and so should be fixed with the latest nightly version of PyTorch. Please try it out and let us know.
